frontend/src/components/payment/paypal-payment-dialog.js

```diff
@@ -6,25 +6,38 @@ import Dialog, {
   DialogContentText,
   DialogTitle
 } from 'material-ui/Dialog'
+import CheckBox from 'material-ui/CheckBox'
 
 class PaypalPaymentDialog extends Component {
+  constructor (props) {
+    super(props)
+    this.agreeTermsPaypal = this.agreeTermsPaypal.bind(this)
+    this.state = { termsPaypal: false }
+  }
+
   componentWillMount () { }
 
   handleNewOrder = (e) => {
     e.preventDefault()
-    this.props.createOrder({
-      provider: 'paypal',
-      currency: 'USD',
-      amount: this.props.itemPrice,
-      userId: this.props.user.id,
-      TaskId: this.props.task
-    })
+    if(this.state.termsPaypal){
+      this.props.createOrder({
+        provider: 'paypal',
+        currency: 'USD',
+        amount: this.props.itemPrice,
+        userId: this.props.user.id,
+        TaskId: this.props.task
+      })
+    }
   }
 
   triggerPayment (order) {
     window.location.href = order.payment_url
   }
 
+  agreeTermsPaypal (){
```

Comment:
```suggestion
  agreeTermsPaypal = () => {
```
This would eliminate the bind on constructor